distributed/worker_memory.py

```diff
@@ -109,9 +118,20 @@ def __init__(
         if isinstance(data, MutableMapping):
             self.data = data
         elif callable(data):
-            self.data = data()
+            if has_arg(data, "worker_local_directory"):
```

Comment:
Unlike the case without arguments, which you can figure out by just reading the signature, there is no way to know how to use this without reading the implementation code. Please document.

Reply:
Added `__init__` docstring with usage note (mostly referring to `Worker.__init__` for the meaning of all of the other parameters).